algorithms/searches/ExponentialSearch.rs

```diff
@@ -0,0 +1,51 @@
+mod search {
+    pub fn exponential_search(array: &[i32], item: i32) -> i32 {
+        if array.len() == 0 {
+            return -1;
+        }
+        let mut i = 1;
+        while i < array.len() {
+            if array[i] < item {
+                i *= 2;
+            } else {
+                break;
+            }
+        }
+        if i >= array.len() {
+            i = array.len() - 1;
+        }
+        let index = binary_search(&array[(i/2)..=i], item);
+        if index != -1 { index + (i as i32 / 2) } else { -1 }
+    }

```

Comment:
Exponential search includes the Binary search itself. Code for the Binary search I take from `BinarySearch.rs`